visual_marker_mapping/include/visual_marker_mapping/TagDetector.h

```diff
@@ -37,9 +37,14 @@ struct DetectionResult
 };
 
 DetectionResult detectTags(const std::string& folder, double markerWidth, double markerHeight,
-    const std::string& tagType = "apriltag_36h11", bool doCornerRefinement = false);
+    const std::string& tagType = "apriltag_36h11", bool doCornerRefinement = false,
+    bool umich = false);
```

Comment:
Ich wuerde vorschlagen eine DetectionResults.h zu machen. Dort kommen die ganzen Datenstrukturen hin (TagDetection, usw...). Dann machst du einen TagDetector_MIT.{cpp.h} und packst dort die MIT Implementierung hin. Analog fuer umich. Die dort bereitgestellten Funktionen haben den selben Namen sind aber in einem anderen Namespace innerhalb des visual_marker_mapping namespaces. Ein switchen nach dem Verfahren erfolgt dann aus der main Funktion raus.